### PR TITLE
 #1637: Add servant/tt_001.bin

### PR DESCRIPTION
**Pull request by JoshSchreuder**
Followed [the guide](https://github.com/Xeeynamo/sotn-decomp/wiki/Decompilation#add-new-overlay) which seemed to work relatively painlessly for this overlay.

I hit an issue in `make-config.py` where no duplicates were found and have added a separate commit which fixed that. Otherwise no issues with guide.
```
josh@JoshsPC:/mnt/c/dev/sotn-decomp$ tools/make-config.py tt_001
✔ generating psx splat config
✔ splitting config/splat.us.tt_001.yaml
✔ adjusting files at src/servant/tt_001
✔ disassembling matched functions
✖ finding duplicates across overlays
Traceback (most recent call last):
  File "/mnt/c/dev/sotn-decomp/tools/make-config.py", line 1132, in <module>
    raise e
  File "/mnt/c/dev/sotn-decomp/tools/make-config.py", line 1126, in <module>
    make_config(args.input, args.version)
  File "/mnt/c/dev/sotn-decomp/tools/make-config.py", line 1106, in make_config
    if found > 0:
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -489,6 +498,7 @@ disk_prepare: build $(SOTNDISK)
 	cp $(BUILD_DIR)/MAR.BIN $(DISK_DIR)/BOSS/MAR/MAR.BIN
 	cp $(BUILD_DIR)/F_MAR.BIN $(DISK_DIR)/BOSS/MAR/F_MAR.BIN
 	cp $(BUILD_DIR)/TT_000.BIN $(DISK_DIR)/SERVANT/TT_000.BIN
+	cp $(BUILD_DIR)/TT_001.BIN $(DISK_DIR)/SERVANT/TT_001.BIN
 disk: disk_prepare
 	$(SOTNDISK) make build/sotn.$(VERSION).cue $(DISK_DIR) $(CONFIG_DIR)/disk.us.lba
 disk_debug: disk_prepare
```
Comment (JoshSchreuder) [inline Makefile:504]: `disk_debug` does something with `TT_000.BIN` which I was unclear on, do I need to add anything in there too for `TT_001.BIN`?

File: README.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ This repo does not include any assets or assembly code necessary for compiling t
...
 | BOSS/MAR/MAR.BIN | ![code coverage MAR.BIN](https://img.shields.io/endpoint?label=MAR%20code&url=https%3A%2F%2Fprogress.deco.mp%2Fdata%2Fsotn%2Fus%2Fcode%2F%3Fmode%3Dshield%26measure%3Dbomar) | ![decompiled functions](https://img.shields.io/endpoint?label=MAR%20funcs&url=https%3A%2F%2Fprogress.deco.mp%2Fdata%2Fsotn%2Fus%2Ffunctions%2F%3Fmode%3Dshield%26measure%3Dbomar) | Maria cutscene
 | SERVANT/TT_000.BIN | ![code coverage TT_000.BIN](https://img.shields.io/endpoint?label=TT_000%20code&url=https%3A%2F%2Fprogress.deco.mp%2Fdata%2Fsotn%2Fus%2Fcode%2F%3Fmode%3Dshield%26measure%3Dtt_000) | ![decompiled functions](https://img.shields.io/endpoint?label=TT_000%20funcs&url=https%3A%2F%2Fprogress.deco.mp%2Fdata%2Fsotn%2Fus%2Ffunctions%2F%3Fmode%3Dshield%26measure%3Dtt_000) | Bat Familiar
+| SERVANT/TT_001.BIN | ![code coverage TT_001.BIN](https://img.shields.io/endpoint?label=TT_000%20code&url=https%3A%2F%2Fprogress.deco.mp%2Fdata%2Fsotn%2Fus%2Fcode%2F%3Fmode%3Dshield%26measure%3Dtt_001) | ![decompiled functions](https://img.shields.io/endpoint?label=TT_000%20funcs&url=https%3A%2F%2Fprogress.deco.mp%2Fdata%2Fsotn%2Fus%2Ffunctions%2F%3Fmode%3Dshield%26measure%3Dtt_001) | Familiar 2
```
Comment (JoshSchreuder) [inline README.md:34]: Wasn't sure what this familiar actually is - I did some searching around and didn't find anything, I can update this if you let me know

File: tools/make-config.py [hunk 3]
```diff
@@ -1103,12 +1103,12 @@ def make_config(ovl_name: str, version: str):
         hydrate_psp_symbols(splat_config_path, splat_config, version)
     else:
         found = hydrate_psx_duplicate_symbols(splat_config, ovl_name, version)
-        if found > 0:
+        if found is not None and found > 0:
```
Comment (JoshSchreuder) [inline tools/make-config.py:1106]: I think this `None` is coming from https://github.com/JoshSchreuder/sotn-decomp/blob/6c28f88ec14843f16c00aeb3a55ca4ef1db14743/tools/make-config.py#L980 ?

Would you prefer if I update that to return 0 instead?

File: README.md (hunk 2 above)
Comment (sozud) [reply README.md:34]: It ought to be the ghost https://github.com/Xeeynamo/sotn-decomp/blob/d43a7aeace80247131bc5244369d704d0b3f76cc/include/game.h#L1040

File: Makefile (hunk 1 above)
Comment (sozud) [reply Makefile:504]: We have the option of replacing the bat overlay with a debugging overlay instead. I don't think you need to make changes for it https://github.com/Xeeynamo/sotn-decomp/tree/master/tools/sotn-debugmodule

**Review by sozud — APPROVED** [review]

File: tools/dups/src/main.rs [hunk 4]
```diff
@@ -244,6 +244,13 @@ fn do_dups_report(output_file: Option<String>, threshold: f64) {
             include_asm: get_all_include_asm("../../src/servant/tt_000"),
             path_matcher: "/tt_000/".to_string(),
         },
+        SrcAsmPair {
+            asm_dir: String::from("../../asm/us/servant/tt_001/nonmatchings/"),
+            src_dir: String::from("../../src/servant/tt_001"),
+            overlay_name: String::from("RIC"),
```
[the remark this answers is not in the document]
Comment (JoshSchreuder) [reply tools/dups/src/main.rs]: Fixed and rebased off your changes, thanks.

**Pull request merged**